src/ape/api/providers.py

```diff
@@ -614,6 +634,11 @@ def get_transaction(self, txn_hash: str, required_confirmations: int = 0) -> Rec
         )
         return receipt.await_confirmations()
 
+    def get_transactions_by_block(self, block_id: HexBytes) -> Iterator[TransactionAPI]:
+        block = self.provider.get_block(block_id)
```

Comment:
May need to pass-through `full_transactions` boolean to `.get_block`

Reply:
not quite sure how to do this yet, will review

Reply:
This is how Web3py exposes this feature. Our `ProviderAPI` implementations use it under the hood:
https://web3py.readthedocs.io/en/stable/web3.eth.html#web3.eth.Eth.get_block

Reply:
From the `ProviderAPI` perspective, the boolean makes sense I think